### PR TITLE
Use their new JWTs

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -50,11 +52,41 @@ var tsclient *tailscale.LocalClient
...
+	log.Println("Logging in to Mosyle ...")
+	err := mosyleLogin()
+	if err != nil {
+		log.Fatalf("failed to login to mosyle: %v", err)
+	}
+
+	go func() {
+	DailyRefresh:
+		for {
+			time.Sleep(23 * time.Hour)
```
Comment (cole-h) [inline main.go:64]: Would be nice to get more than 1 hour of leeway, but maybe it doesn't matter

```suggestion
			time.Sleep(20 * time.Hour)
```

File: main.go [hunk 2]
```diff
@@ -50,11 +52,41 @@ var tsclient *tailscale.LocalClient
...
+
+			err := mosyleLogin()
+			if err == nil {
+				continue DailyRefresh
+			}
+
+			log.Printf("failed to login to mosyle, trying again several more times: %v\n", err)
+			for i := 0; i <= 55; i += 1 {
+				time.Sleep(55 * time.Second)
```
Comment (cole-h) [inline main.go:73]: Why 55?

File: main.go [hunk 3]
```diff
@@ -50,11 +52,41 @@ var tsclient *tailscale.LocalClient
...
+				if err == nil {
+					continue DailyRefresh
+				}
+				log.Printf("Failure #%d/55: %v\n", i, err)
+			}
+			log.Fatalf("No luck after a bunch of attempts")
+		}
+	}()
+
+	log.Fatalf("Verifying we can fetch machines from Mosyle...")
```
Comment (cole-h) [inline main.go:84]: > Fatalf is equivalent to [Printf](https://pkg.go.dev/log#Printf) followed by a call to [os.Exit](https://pkg.go.dev/os#Exit)(1).

File: main.go [hunk 4]
```diff
@@ -238,6 +270,51 @@ func getDeviceFromName(devices []Device, name string) (*Device, error) {
...
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if res.StatusCode == 200 {
+		if len(res.Header["Authorization"]) == 1 {
+			jwt = res.Header["Authorization"][0]
```
Comment (cole-h) [inline main.go:305]: Does the JWT not need `Bearer ...`, or is that returned as part of the response?

File: main.go (hunk 2 above)
Comment (grahamc) [reply main.go:73]: each attempt takes ~ a while, so I figured I'd just make it a little less than a full minute. I dunno :)

File: main.go (hunk 4 above)
Comment (grahamc) [reply main.go:305]: it is part of the response

File: main.go (hunk 2 above)
Comment (cole-h) [reply main.go:73]: Let's just make it a full minute so that it's less "magical" seeming

Comment (grahamc) [inline main.go:73]:
```suggestion
			for i := 0; i <= 60; i += 1 {
				time.Sleep(220 * time.Second)
```

Comment (cole-h) [reply main.go]: Update the `#%d/55` line below too

**Review by cole-h — APPROVED** [review]

**Pull request merged**